src/azure-cli-core/azure/cli/core/commands/arm.py

```diff
@@ -178,8 +178,20 @@ def __call__(self, parser, namespace, values, option_string=None):
                 (dest) fields will also be of type `IterateValue`
                 '''
                 from msrestazure.tools import parse_resource_id
+                import os
+                if isinstance(values, str):
+                    values = [values]
+                expanded_values = []
+                for val in values:
+                    try:
+                        # support piping values from JSON. Does not require use of --query
+                        json_val = json.loads(val)
+                        expanded_values = expanded_values + [x['id'] for x in json_val]
+                    except ValueError:
+                        # supports piping of --ids to the command when using TSV. Requires use of --query
+                        expanded_values = expanded_values + val.split(os.linesep)
```

Comment:
?

Reply:
Changed this based on our discussion. If you supply JSON, it ignores any objects that don't have a top-level ID field.